docs/extending.md

```diff
@@ -1,208 +1,296 @@
-
-```java
-@Implements(Bitmap.class)
-public class MyShadowBitmap {
-  @RealObject private Bitmap realBitmap;
-  private int bitmapQuality = -1;
-    
-  @Implementation
-  public boolean compress(Bitmap.CompressFormat format, int quality, OutputStream stream) {
-    bitmapQuality = quality;
-    return realBitmap.compress(format, quality, stream);
-  }
-
-  public int getQuality() {
-    return bitmapQuality;
-  }
-}
-```
+Custom shadows are structured much the same as normal shadow classes. They must include the `@Implements`
+annotation on the class definition. You can use the normal shadow implementation options, such as shadowing instance
+methods using `@Implementation` or shadowing constructors using `__constructor__()` methods. Your shadow class may also extend one of the stock Robolectric shadows if you like.
+
+=== "Java"
+
+    ```java
+    @Implements(Bitmap.class)
+    public class MyShadowBitmap {
+      @RealObject private Bitmap realBitmap;
+      private int bitmapQuality = -1;
+
+      @Implementation
+      public boolean compress(Bitmap.CompressFormat format, int quality, OutputStream stream) {
+        bitmapQuality = quality;
+        return realBitmap.compress(format, quality, stream);
+      }
+
+      public int getQuality() {
+        return bitmapQuality;
+      }
+    }
+    ```
+
+=== "Kotlin"
+
+    ```kotlin
+    @Implements(Bitmap::class)
+    class MyShadowBitmap {
+      @RealObject private lateinit var realBitmap: Bitmap
+      private var bitmapQuality: Int = -1;
+
+      @Implementation
+      fun compress(format: Bitmap.CompressFormat, quality: Int, stream: OutputStream): Boolean {
+        bitmapQuality = quality
+        return realBitmap.compress(format, quality, stream)
+      }
+
+      fun getQuality(): Int {
+        return bitmapQuality
+      }
+    }
+    ```
 
 ### Using a Custom Shadows
 
-Custom Shadows get hooked up to Robolectric using the `@Config` annotation on the test class or test method, using the `shadows` array attribute.  To use the `MyShadowBitmap` class mentioned in the previous section, you would annotate the test in question with `@Config(shadows={MyShadowBitmap.class})`, and to include multiple custom shadows: `@Config(shadows={MyShadowBitmap.class, MyOtherCustomShadow.class})`.  This causes Robolectric to recognize and use your custom shadow when executing code against the class you shadowed.
+Custom Shadows get hooked up to Robolectric using the [`@Config`](javadoc/latest/org/robolectric/annotation/Config.html) annotation on the test class or test method, using the [`shadows`](javadoc/latest/org/robolectric/annotation/Config.html#shadows()) array attribute. To use the `MyShadowBitmap` class mentioned in the previous section, you would annotate the test in question with `@Config(shadows = { MyShadowBitmap.class })`. This causes Robolectric to recognize and use your custom shadow when executing code against the class you shadowed.
```

Comment:
I've removed the example with multiple custom shadows, as it teaches more about Java syntax rather than Robolectric usage.